server/odc-service/src/test/java/com/oceanbase/odc/service/sqlcheck/MySQLCheckerTest.java

```diff
@@ -1362,6 +1362,107 @@ public void check_wrongPlExists_violationGenerated() {
         Assert.assertEquals(expect, actual);
     }
 
+    @Test
+    public void check_restrictSqlAffectedRows4MySQL_violationGenerated() {
+        String insert =
+                "insert into users (id, name, age, email) values "
+                        + "('2', 'b-bot', 3, 'o'),"
+                        + "('3', 'c-bot', 3, 'o'),"
+                        + "('4', 'd-bot', 3, 'o'),"
+                        + "('5', 'e-bot', 3, 'o')";
+        String update =
+                "update users set name = 'a1-bot' where id = '1'";
+        String delete =
+                "delete from users where id in ('1', '2', '3')";
+        String select =
+                "select id, name, age from users where id in ('1', '2')";
+
+        JdbcTemplate jdbcTemplate = Mockito.mock(JdbcTemplate.class);
+
+        Mockito.when(jdbcTemplate.query(Mockito.anyString(), Mockito.any(RowMapper.class)))
+                .thenReturn(Collections.singletonList(4L));
+        DefaultSqlChecker insertChecker = new DefaultSqlChecker(DialectType.MYSQL, "$$",
+                Collections.singletonList(
+                        new MySQLAffectedRows(2, DialectType.MYSQL, jdbcTemplate)));
+        List<CheckViolation> actualInsert = insertChecker.check(insert);
+        Assert.assertEquals(1, actualInsert.size());
+
+        Mockito.when(jdbcTemplate.query(Mockito.anyString(), Mockito.any(RowMapper.class)))
+                .thenReturn(Collections.singletonList(1L));
+        DefaultSqlChecker updateChecker = new DefaultSqlChecker(DialectType.MYSQL, "$$",
+                Collections.singletonList(
+                        new MySQLAffectedRows(2, DialectType.MYSQL, jdbcTemplate)));
+        List<CheckViolation> actualUpdate = updateChecker.check(update);
+        Assert.assertEquals(0, actualUpdate.size());
+
+        Mockito.when(jdbcTemplate.query(Mockito.anyString(), Mockito.any(RowMapper.class)))
+                .thenReturn(Collections.singletonList(3L));
+        DefaultSqlChecker deleteChecker = new DefaultSqlChecker(DialectType.MYSQL, "$$",
+                Collections.singletonList(
+                        new MySQLAffectedRows(2, DialectType.MYSQL, jdbcTemplate)));
+        List<CheckViolation> actualDelete = deleteChecker.check(delete);
+        Assert.assertEquals(1, actualDelete.size());
+
+        Mockito.when(jdbcTemplate.query(Mockito.anyString(), Mockito.any(RowMapper.class)))
+                .thenReturn(Collections.singletonList(2L));
+        DefaultSqlChecker selectChecker = new DefaultSqlChecker(DialectType.MYSQL, "$$",
+                Collections.singletonList(
+                        new MySQLAffectedRows(2, DialectType.MYSQL, jdbcTemplate)));
+        List<CheckViolation> actualSelect = selectChecker.check(select);
+        Assert.assertEquals(0, actualSelect.size());
+    }
+
+    @Test
+    public void check_restrictSqlAffectedRows4OBMySQL_violationGenerated() {
```

Comment:
one case one method, may split into cases

Reply:
OK, the style is  follow with other rules.

Reply:
OK, the style is  follow with other rules.